gapic/src/main/com/google/gapid/widgets/ImagePanel.java

```diff
@@ -672,25 +716,37 @@ public void init(Renderer renderer) {
 
     @Override
     public void update(Renderer renderer, SceneData newData) {
-      Image oldImage = (this.data != null) ? this.data.image : null;
-      if (oldImage != newData.image) {
-        // Release old texture, create new.
-        if (texture != null) {
-          texture.delete();
+      // Release textures that are no longer in data.
+      Set<Image> newSet = Sets.newHashSet(newData.images);
+      for (Entry<Image, Texture> entry : imageToTexture.entrySet()) {
```

Comment:
nit: use `Map.Entry` and remove the import. `Entry` is so generic and `Map.Entry` so short, that I prefer it that way.

Reply:
Done